File: backend/app/routes/what_if.py
from fastapi import APIRouter
from app.models.what_if import WhatIfRequest, WhatIfResponse
from app.services.what_if_service import what_if_service

router = APIRouter(prefix="/api/what-if", tags=["what-if"])

@router.post("", response_model=WhatIfResponse)
async def what_if_endpoint(request: WhatIfRequest):
    return what_if_service.execute_scenario(request)

File: backend/src/routes/whatIf.js
import express from "express";
import { whatIfService } from "../services/whatIfService.js";

const router = express.Router();

router.post("/", (req, res) => {
  try {
    const result = whatIfService.executeScenario(req.body);
    return res.json(result);
  } catch (err) {
    console.error("What-If route error:", err);
    return res.status(500).json({ error: "Failed to process What-If scenario" });
  }
});

export default router;
